diff --git a/Makefile.am b/Makefile.am
index 10b9670..7104647 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS=dbus bus tools test doc
+SUBDIRS=dbus bus 
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = dbus-1.pc
diff --git a/dbus/Makefile.am b/dbus/Makefile.am
index bb5ccca..dcf171e 100644
--- a/dbus/Makefile.am
+++ b/dbus/Makefile.am
@@ -114,16 +114,18 @@ DBUS_UTIL_arch_sources = 			\
 endif
 
 if HAVE_LINUX_EPOLL
-DBUS_UTIL_arch_sources += dbus-socket-set-epoll.c
+DBUS_SHARED_arch_sources += dbus-socket-set-epoll.c
 endif
 
 dbusinclude_HEADERS=				\
 	dbus.h					\
 	dbus-address.h				\
 	dbus-bus.h				\
+	dbus-ccsp-apis.h			\
 	dbus-connection.h			\
 	dbus-errors.h				\
 	dbus-macros.h				\
+	dbus-mainloop.h				\
 	dbus-memory.h				\
 	dbus-message.h				\
 	dbus-misc.h				\
@@ -147,6 +149,7 @@ DBUS_LIB_SOURCES=				\
 	dbus-auth.c				\
 	dbus-auth.h				\
 	dbus-bus.c				\
+	dbus-ccsp-apis.c			\
 	dbus-connection.c			\
 	dbus-connection-internal.h		\
 	dbus-credentials.c			\
@@ -212,6 +215,8 @@ DBUS_SHARED_SOURCES=				\
 	dbus-internals.h			\
 	dbus-list.c				\
 	dbus-list.h				\
+	dbus-mainloop.c				\
+	dbus-mainloop.h				\
 	dbus-marshal-basic.c			\
 	dbus-marshal-basic.h			\
 	dbus-memory.c				\
@@ -219,6 +224,9 @@ DBUS_SHARED_SOURCES=				\
 	dbus-mempool.h				\
 	dbus-pipe.c                 \
 	dbus-pipe.h                 \
+	dbus-socket-set.h			\
+	dbus-socket-set.c			\
+	dbus-socket-set-poll.c			\
 	dbus-string.c				\
 	dbus-string.h				\
 	dbus-string-private.h			\
@@ -237,8 +245,6 @@ DBUS_UTIL_SOURCES=				\
 	dbus-auth-script.h			\
 	dbus-auth-util.c			\
 	dbus-credentials-util.c			\
-	dbus-mainloop.c				\
-	dbus-mainloop.h				\
 	dbus-marshal-byteswap-util.c		\
 	dbus-marshal-recursive-util.c		\
 	dbus-marshal-validate-util.c		\
@@ -248,9 +254,6 @@ DBUS_UTIL_SOURCES=				\
 	dbus-shell.c				\
 	dbus-shell.h				\
 	$(DBUS_UTIL_arch_sources)		\
-	dbus-socket-set.h			\
-	dbus-socket-set.c			\
-	dbus-socket-set-poll.c			\
 	dbus-spawn.h				\
 	dbus-string-util.c			\
 	dbus-sysdeps-util.c			\
diff --git a/dbus/dbus.h b/dbus/dbus.h
index 932ceab..949a5d4 100644
--- a/dbus/dbus.h
+++ b/dbus/dbus.h
@@ -29,9 +29,11 @@
 #include <dbus/dbus-arch-deps.h>
 #include <dbus/dbus-address.h>
 #include <dbus/dbus-bus.h>
+#include <dbus/dbus-ccsp-apis.h>
 #include <dbus/dbus-connection.h>
 #include <dbus/dbus-errors.h>
 #include <dbus/dbus-macros.h>
+#include <dbus/dbus-mainloop.h>
 #include <dbus/dbus-message.h>
 #include <dbus/dbus-misc.h>
 #include <dbus/dbus-pending-call.h>
diff --git a/m4/tp-compiler-warnings.m4 b/m4/tp-compiler-warnings.m4
index ee4af31..2a890ae 100644
--- a/m4/tp-compiler-warnings.m4
+++ b/m4/tp-compiler-warnings.m4
@@ -16,7 +16,7 @@ AC_DEFUN([TP_COMPILER_WARNINGS],
     TP_COMPILER_FLAG([-W$tp_flag], [tp_warnings="$tp_warnings -W$tp_flag"])
   done
 
-  tp_error_flags="-Werror"
+  tp_error_flags=""
   TP_COMPILER_FLAG([-Werror], [tp_werror=yes], [tp_werror=no])
 
   for tp_flag in $4; do
